mlrun/db/httpdb.py

```diff
@@ -1142,7 +1142,23 @@ def delete_function(self, name: str, project: str = ""):
         project = project or config.default_project
         path = f"projects/{project}/functions/{name}"
         error_message = f"Failed deleting function {project}/{name}"
-        self.api_call("DELETE", path, error_message)
+        response = self.api_call("DELETE", path, error_message, version="v2")
+        if response.status_code == http.HTTPStatus.ACCEPTED:
+            logger.info(
+                "Function is being deleted", project_name=project, function_name=name
+            )
+            background_task = mlrun.common.schemas.BackgroundTask(**response.json())
+            background_task = self._wait_for_background_task_to_reach_terminal_state(
+                background_task.metadata.name
+            )
```

Comment:
```suggestion
            background_task = self._wait_for_background_task_to_reach_terminal_state(
                background_task.metadata.name, project=project
            )
```
So it will get the project background task and not the internal one